pkg/networking/networking.go

```diff
@@ -101,19 +122,21 @@ func (n *networkImpl) addDefaultHeader(request *http.Request) {
 	}
 }
 
-func (n *networkImpl) GetRoundTripper() http.RoundTripper {
-	transport := n.configureRoundTripper(http.DefaultTransport.(*http.Transport))
-
-	rt := middleware.NewAuthHeaderMiddleware(n.config, n.GetAuthenticator(), transport)
-
-	// encapsulate everything
-	roundTrip := customRoundTripper{
+func (n *networkImpl) addMiddlewaresToRoundTripper(rt http.RoundTripper) http.RoundTripper {
```

Comment:
This changes the API we should not change exported functions of packages that are not in `internal`